pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -50,26 +66,22 @@ const logStreamIdentifierInternal = "__logstream__grafana_internal__"
 var plog = log.New("tsdb.cloudwatch")
 var aliasFormat = regexp.MustCompile(`\{\{\s*(.+?)\s*\}\}`)
 
-func init() {
-	tsdb.RegisterTsdbQueryEndpoint("cloudwatch", func(ds *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
-		return newExecutor(), nil
-	})
-}
+const defaultConcurrentQueries = 4
 
 func newExecutor() *cloudWatchExecutor {
 	return &cloudWatchExecutor{
 		logsClientsByRegion: map[string]cloudwatchlogsiface.CloudWatchLogsAPI{},
+		queuesByRegion:      map[string]chan bool{},
 	}
 }
 
-// cloudWatchExecutor executes CloudWatch requests.
-type cloudWatchExecutor struct {
-	*models.DataSource
+func init() {
+	globalExecutor := newExecutor()
+	tsdb.RegisterTsdbQueryEndpoint("cloudwatch", func(ds *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
+		return globalExecutor, nil
+	})
```

Comment:
I would revert this change, since it only makes the code more complicated. I'm also wondering why the function is moved, since it makes the diff larger.

Reply:
This change is necessary since otherwise a new executor is created for each query request, which would mean query state isn't persisted between queries

Reply:
@kaydelaney Ah, I didn't consider that it might be intentional to have a persisted closure variable.